src/components/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,343 @@
+
+  // a default date validation that peforms basic checks
+  const defaultErrorValidation = (stateDateObj: any) => {
+    if (stateDateObj.mm && (stateDateObj.mm == 0 || stateDateObj.mm > 12)) {
+      setDateError("Invalid Date");
+      return;
+    }
+    if (
+      (stateDateObj.yyyy && stateDateObj.yyyy.length < 4) ||
+      ["0000", "9999"].includes(stateDateObj.yyyy)
+    ) {
+      setDateError("Invalid Date");
+      return;
+    }
+    let availableMonths = daysInMonth(stateDateObj.mm, stateDateObj.yyyy);
+    if (
+      stateDateObj.dd &&
+      (stateDateObj.dd == 0 || stateDateObj.dd > availableMonths)
+    ) {
+      setDateError("Invalid Date");
+      return;
+    }
+    // if no errors then set the error state as empty
+    setDateError("");
+  };
+
+  // Function to get number of days available in a mont
+  function daysInMonth(month: number, year: number) {
+    return new Date(year, month, 0).getDate();
+  }
+
+  function handleBackSpace(currentIndex: number, event: any) {
+    // if there is a field before the current index then delete and focus on that index
+    if (currentIndex >= 0 && !event.target.value) {
+      let prevField = document.getElementById(
+        `date-input-${currentIndex - 1}-${id}`
+      );
+      prevField?.focus();
+    }
+    defaultErrorValidation(date);
+  }
+
+  //function to handle backspace event
+  function handleKeyDown(event: any, currentIndex: number) {
+    if ([8, 46].includes(event.keyCode)) {
+      handleBackSpace(currentIndex, event);
+    }
+  }
+
+  const handleDateClicked = (dateValue: string) => {
+    setDatePickerDate(dateValue);
+    let dateVal = dateValue.split("/");
+    let stateDateObj = { ...date };
+    stateDateObj.mm = dateVal[0];
+    stateDateObj.dd = dateVal[1];
+    stateDateObj.yyyy = dateVal[2];
+    setDate(stateDateObj);
+    getDateValue?.(
+      new Date(`${stateDateObj.mm}/${stateDateObj.dd}/${stateDateObj.yyyy}`)
+    );
+  };
+
+  return (
+    <div className="n-date-wrapper">
+      <div className={`n-input-label-container`}>
+        {label && (
+          <label className="n-input-label" data-testid={`n-date-label`}>
+            <>
+              {label} {required ? " *" : ""}
+            </>
+          </label>
+        )}
+      </div>
+      <div
+        data-testid={`n-date-inputfield`}
+        className={`n-date-inputfield ${
+          dateError
+            ? "n-date-error"
+            : validationState
+            ? `n-${validationState}-field`
+            : ""
+        }`}
+      >
+        <div
+          className="n-date-left-group"
+          data-testid={`n-date-left-group`}
+          onClick={() =>
+            useDatePicker ? setShowPicker(!showPicker) : () => {}
+          }
+        >
+          <div
+            data-testid={`n-datepicker-icon`}
+            className={`n-icon-container ${
+              useDatePicker ? "n-cursor-pointer" : ""
+            }`}
+            onClick={useDatePicker ? () => {} : () => {}}
```

Comment:
whether useDatePicker is true or false, the function assigned to onClick will be (() => {}).

Reply:
Changes made 👍